client/platform/desktop/backend/ipcService.ts

```diff
@@ -1,57 +1,64 @@
 import OS from 'os';
-
+import http from 'http';
 import { ipcMain } from 'electron';
-import { DesktopJobUpdate, RunPipeline, Settings } from '../constants';
 
-import server from './server';
-import linux from './platforms/linux';
-import win32 from './platforms/windows';
-import common from './platforms/common';
+import { DesktopJobUpdate, RunPipeline, Settings } from 'platform/desktop/constants';
 
-export default function register() {
-  ipcMain.handle('info', () => {
-    const addr = server.address();
-    return addr;
-  });
+import linux from './native/linux';
+import win32 from './native/windows';
+import common from './native/common';
+import settings from './state/settings';
+import { listen } from './server';
 
+// defaults to linux if win32 doesn't exist
+const currentPlatform = OS.platform() === 'win32' ? win32 : linux;
+if (OS.platform() === 'win32') {
+  win32.initialize();
+}
+
+export default function register() {
   /**
    * Platform-agnostic methods
    */
-
-  ipcMain.handle('get-pipeline-list', async (_, settings: Settings) => {
-    const ret = await common.getPipelineList(settings);
+  ipcMain.handle('server-info', async () => {
+    const server = await new Promise<http.Server>((resolve) => listen(resolve));
+    return server.address();
+  });
+  ipcMain.handle('get-pipeline-list', async () => {
+    const ret = await common.getPipelineList(settings.get());
     return ret;
   });
   ipcMain.handle('open-link-in-browser', (_, url: string) => {
     common.openLink(url);
   });
+  ipcMain.on('update-settings', async (_, s: Settings) => {
+    settings.set(s);
+  });
+  ipcMain.handle('import-media', async (_, path: string) => {
+    const ret = await common.importMedia(settings.get(), path);
+    return ret;
+  });
```

Comment:
This just here for possible future uses?  It is never called now that we are using the REST version.

Reply:
Duplicate, should be removed.